src/desktop_pixmap.cpp

```diff
@@ -59,35 +62,59 @@ desktop_pixmap::save_screen (pixmap_format pixmap_format)
   // Not freeing gi_reply causes memory leak,
   // as xcb_get_image always allocates new space
   // This deallocates saved image on x server
-  delete[] this->gi_reply;
-
-  auto gi_cookie = xcb_get_image (
-      drawable::c_,            /* Connection */
-      this->format,            /* Image format */
-      drawable::screen_->root, /* Screenshot relative to root */
-      this->x, this->y, this->width, this->height, /* Dimensions */
-      static_cast<uint32_t> (~0) /* Plane mask (all bits to get all planes) */
-  );
-
-  // TODO Handle errors
-  // Saving reply to free it later. Fixes memory leak
-  this->gi_reply = xcb_get_image_reply (drawable::c_, gi_cookie, nullptr);
-
-  // Casting for later use
-  this->length = static_cast<uint32_t> (xcb_get_image_data_length (gi_reply));
-  this->image_ptr = xcb_get_image_data (gi_reply);
-}
+  uint16_t screen_width = 1920;  // TODO: from conf
+  uint16_t screen_height = 1080; // TODO: from conf
+  uint16_t resized_width = 800;  // TODO: from conf
+  // resized height!!!!
+  uint16_t max_height = MAX_MALLOC / screen_width / 4;
+  uint16_t y_exception; // max_height might be changed in last iteration,
```

Comment:
Please comment the purpose of these variables, as their name don't convey any information.

Reply:
y_exception(it was kinda hard to come up with the name) is needed instead of just (i*height) in functions like xcb_get_image because the last iteration may get a piece of picture smaller than all the previous, e.g. we have an image of five rows and max_malloc is 2 rows: first two iterations work fine, but the last one must handle 1 row, not two. y_exception works this situation out.

Reply:
Hmm, i think a brief algorithm explanation in the function's docstring will help